c/tskit/tables.c

```diff
@@ -10718,14 +10721,14 @@ tsk_table_collection_write_format_data(const tsk_table_collection_t *self,
     /* This stupid dance is to workaround the fact that compilers won't allow
      * casts to discard the 'const' qualifier. */
     tsk_memcpy(format_name, TSK_FILE_FORMAT_NAME, sizeof(format_name));
-    ret = write_table_cols(store, write_cols, 0);
+    ret = write_table_cols(store, write_cols, options, 0);
```

Comment:
This is the only place that we're **not** calling write_table_cols with the BORROW option, so I think it would be simpler to write a different function ``write_table_const_cols`` or something, which passes in the const values and avoids the stupid dance that the comment above here is complaining about.

Everywhere else is doing KAS_BORROWS_ARRAY right?

Reply:
I think so - so the idea would be to not pass kas_flags at all and use the const status to determine if we borrow?

Reply:
Yeah - we borrow everything in the current code path (except the case you've identified where we're casting), and make a new code path to accept the const values where we don't borrow.

Reply:
Sorry, I'm confused here.
The arrays that are passed to kastore are all const as defined in `write_table_col_t`. In fact I just deleted the "stupid dance" (now in this PR) without changing anything else and nothing seems broken.